app/src/features/workspaces/hooks/useFetchLocalWorkspaces.ts

```diff
@@ -0,0 +1,70 @@
+import { captureException } from "@sentry/react";
+import APP_CONSTANTS from "config/constants";
+import { useCheckLocalSyncSupport } from "features/apiClient/helpers/modules/sync/useCheckLocalSyncSupport";
+import { Workspace, WorkspaceMemberRole } from "features/workspaces/types";
+import { useCallback, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { getAllWorkspaces } from "services/fsManagerServiceAdapter";
+import { getUserAuthDetails } from "store/slices/global/user/selectors";
+import { WorkspaceType } from "types";
+import { submitAttrUtil } from "utils/AnalyticsUtils";
+
+export const useFetchLocalWorkspaces = () => {
+  const user = useSelector(getUserAuthDetails);
+  const isLocalSyncEnabled = useCheckLocalSyncSupport({ skipWorkspaceCheck: true });
+
+  const [localWorkspaces, setLocalWorkspaces] = useState<Workspace[]>([]);
+
+  const fetchLocalWorkspaces = useCallback(async () => {
+    if (!isLocalSyncEnabled) {
+      setLocalWorkspaces([]);
+      return;
+    }
+
+    // TODO: uid is needed as per current implementation, to be removed when logged out support is implemented
+    if (!user.loggedIn) {
+      setLocalWorkspaces([]); // TODO: to be removed when local first support is added
+      return;
+    }
+
+    try {
+      const uid = user.details?.profile?.uid;
+
+      const allLocalWorkspacesResult = await getAllWorkspaces();
+      const allLocalWorkspaces = allLocalWorkspacesResult.type === "success" ? allLocalWorkspacesResult.content : [];
+
+      const localRecords = [];
+
+      for (const partialWorkspace of allLocalWorkspaces) {
+        const localWorkspace = {
+          id: partialWorkspace.id,
+          name: partialWorkspace.name,
+          owner: uid,
+          accessCount: 1,
+          adminCount: 1,
+          members: {
```

Comment:
A computed property name must be of type 'string', 'number', 'symbol', or 'any'.ts(2464)